### PR TITLE
RavenDB-12748 added display name support

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/Parser/QueryParser.cs [hunk 1]
```diff
@@ -1975,8 +1975,26 @@ private List<QueryExpression> ReadMethodArguments()
...
                 if (args.Count != 0)
+                {
+                    if (Scanner.TryScan("as"))
+                    {
+                        if (Scanner.Identifier() == false)
+                            ThrowInvalidQueryException("Missing alias for method after 'as'");
+
+                        var lastArg = args[args.Count - 1];
+                        if (lastArg.Type != ExpressionType.Method)
+                            ThrowInvalidQueryException("Alias can only be applied on method");
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/Parser/QueryParser.cs]: Add the alias name here, to make the error better

File: src/Raven.Server/Documents/Queries/Parser/QueryParser.cs [hunk 2]
```diff
@@ -1975,8 +1975,26 @@ private List<QueryExpression> ReadMethodArguments()
                     break;
 
                 if (args.Count != 0)
+                {
+                    if (Scanner.TryScan("as"))
+                    {
+                        if (Scanner.Identifier() == false)
+                            ThrowInvalidQueryException("Missing alias for method after 'as'");
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/Parser/QueryParser.cs]: Add method name here

File: src/Raven.Server/Documents/Queries/Parser/QueryParser.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Queries/Parser/QueryParser.cs]: Done

File: src/Raven.Server/Documents/Queries/Parser/QueryParser.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Queries/Parser/QueryParser.cs]: Done

**Pull request merged**